src/hotspot/cpu/x86/universalUpcallHandler_x86.cpp

```diff
@@ -236,9 +236,9 @@ static address generate_upcall_stub(jobject rec, const ABIDescriptor& abi, const
   return blob->code_begin();
 }
 
-jlong ProgrammableUpcallHandler::generate_upcall_stub(JNIEnv *env, jobject rec, jobject jabi, jobject jlayout) {
-  const ABIDescriptor abi = parseABIDescriptor(env, jabi);
-  const BufferLayout layout = parseBufferLayout(env, jlayout);
+jlong ProgrammableUpcallHandler::generate_upcall_stub(jobject rec, jobject jabi, jobject jlayout) {
+  const ABIDescriptor abi = ForeignGlobals::parseABIDescriptor(jabi);
+  const BufferLayout layout = ForeignGlobals::parseBufferLayout(jlayout);
```

Comment:
These functions should be something like parse_buffer_layout according to the Hotspot coding style.